scripts/tests/test_zkai_attention_kv_fuller_crossing_grid_gate.py

```
@@ -0,0 +1,133 @@
+        self.assertEqual(summary["proved_all_axis_cell_count"], 1)
+        self.assertEqual(summary["missing_all_axis_cell_count"], 15)
+        self.assertEqual(summary["proved_counts_by_width"], {"16": 3, "32": 1, "8": 7})
+        self.assertEqual(summary["proved_counts_by_head_count"], {"1": 3, "16": 1, "2": 5, "4": 1, "8": 1})
+        self.assertEqual(summary["proved_counts_by_steps_per_head"], {"16": 2, "32": 1, "8": 8})
+        self.assertIn("full factorial proof-grid claim", self.result["no_go_criteria"][0])
+        self.assertEqual(self.result["next_low_risk_profiles"][0]["profile_id"], "d32_two_head_seq8")
+
+    def test_validate_rejects_overclaims_and_metric_smuggling(self):
+        bad = copy.deepcopy(self.result)
+        bad["claim_boundary"] = "GO_REAL_VALUED_SOFTMAX_PUBLIC_BENCHMARK"
+        with self.assertRaisesRegex(gate.FullerCrossingGridGateError, "result drift for claim_boundary"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)